programs/survival/tputils.sc

```diff
@@ -0,0 +1,283 @@
+    );
+
+    global_names = keys(global_wdata);
+    if(global_names~name!=null, print(player,format(str('r %s is already in use. Therefore, updating the warp home to new location. \n',name))));
+    if(global_wdata:name==null,global_wdata:name={});
+
+    global_wdata:name:'location' = pos(player);
+    global_wdata:name:'pitch' = player~'pitch';
+    global_wdata:name:'yaw' = player~'yaw';
+    global_wdata:name:'dimension' = player~'dimension';
+    global_wdata:name:'description'= description;
+
+    //Extra stuff that is only availble to people with fake inv screens
+    if(_screenCheck(),
+        global_wdata:name:'item'= query(player,'holds','mainhand'):0;
+    );
+    _saveWData();
+    print(player,format('e Succesfully saved warp to location with name: '+name));
+
+);
+
+_wremove(name)->(
+
+    player=player();
+    if(player~'permission_level'<global_operator_permission_level,
+        print(player,format('r You do not have the permsission to execute this command. If you believe this to be an error, please contact server admin.'));
+        return();
+    );
+
+    if(global_wdata:name==null,
+        print(player,format('r \nThe specefied location does not exist. Please use warp menu to know about current warps'));
+        return();
+    );
+    delete(global_wdata,name);
+    _saveWData();
+    print(player,format(str('e \n%s was succesfully removed from warp list.',name)));
+
+);
+
+_wtp(name) -> (
+
+    player=player();
+    info = global_wdata:name;
+    if(info==null,
+        print(player,format('r \nThe specefied location does not exist. Please use warp menu to know about current warps'));
+        return();
+    );
+    dimension = info:'dimension';
+    pitch = info:'pitch';
+    yaw = info:'yaw';
+    location = info:'location';
+
+    run(str('execute in %s run tp %s %f %f %f %f %f',dimension,player,location:0,location:1,location:2,yaw,pitch));
+
+);
+
+_wmenu() -> (
+
+    player= player();
+    if(_screenCheck(),
+        warpCount = length(keys(global_wdata));
+        rowCount = ceil(warpCount/9);
+
+        menu= create_screen(player,str('generic_9x%s',rowCount),format('tb Available Warps'),_(screen,player,action,data) -> (
+            
+            if(action=='pickup' || action== 'swap' || action=='quick_move' || action== 'throw' || action== 'pickup_all',
+                itemNbt= inventory_get(screen,data:'slot'):2;
+                itemName= parse_nbt(itemNbt:'display':'Name'):0:'text';
+                _wtp(itemName);
+                close_screen(screen);
+            );
+        ));
+
+        if(screen_property(menu,'open'),
+
+            slot=0;
+            for(keys(global_wdata),
+
+                info = global_wdata:_;
+                iType=info:'item';
+                if(iType==null,iType='barrier');
+                des=info:'description';
+                loc=info:'location';
+                x= floor(loc:0);
+                y= floor(loc:1);
+                z= floor(loc:2);

```

Comment:
This is a completely unrelated point, but this is the same as `[x, y, z] = map(loc, floor(_))`.

Reply:
Oman my whole scarpet life was a lie. I never knew this was possible xD. Imma change it everywhere asap